content/300-guides/050-database/850-using-prisma-with-planetscale.mdx

```diff
@@ -0,0 +1,240 @@
+
+This document discusses the concepts behind using Prisma and PlanetScale, explains the commonalities and differences between PlanetScale and other database providers, and leads you through the process for configuring your application to integrate with PlanetScale.
+
+</TopBlock>
+
+## What is PlanetScale?
+
+PlanetScale uses the [Vitess](https://vitess.io/) database clustering system to provide a MySQL-compatible database platform. Features include:
+
+- **Enterprise scalability.** PlanetScale provides a highly available production database cluster that supports scaling across multiple database servers.
+
+- **Database branches.** PlanetScale allows you to create [branches of your database schema](https://docs.planetscale.com/concepts/branching), so that you can test changes on a development branch before applying them to your production database.
+
+- **Support for [non-blocking schema changes](https://docs.planetscale.com/concepts/nonblocking-schema-changes).** PlanetScale provides a workflow that allows users to update database schemas without locking the database or causing downtime.
+
+## Commonalities with other database providers
+
+Many aspects of using Prisma with PlanetScale are just like using Prisma with any other relational database. You can still:
+
+- model your database with the [Prisma Schema Language](/concepts/components/prisma-schema)
+- use Prisma's existing [`mysql` database connector](/concepts/database-connectors/mysql) in your schema, along with the [connection string PlanetScale provides you](https://docs.planetscale.com/concepts/connection-strings)
+- use [Introspection](/concepts/components/introspection) for existing projects if you already have a database schema in PlanetScale
+- use [`db push`](/concepts/components/prisma-migrate/db-push) to push changes in your schema to the database
+- use [Prisma Client](/concepts/components/prisma-client) in your application to talk to the database server at PlanetScale
+
+## Differences to consider
+
+PlanetScale's branching model and design for scalability means that there are also a number of differences to consider. You should be aware of the following points when deciding to use PlanetScale with Prisma:
+
+- **Referential integrity.** To support scaling across multiple database servers, PlanetScale [does not allow the use of foreign key constraints](https://docs.planetscale.com/tutorials/operating-without-foreign-keys), which are normally used in relational databases to enforce relationships between data in different tables. To maintain these relationships in your data while using PlanetScale with Prisma, you will need to use Prisma's ability to [emulate referential integrity in the Prisma client](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-integrity#handling-the-referential-integrity-in-prisma). For more information, see [How to enable emulation of referential integrity](#how-to-enable-emulation-of-referential-integrity).
```

Comment:
This currently does not mention "referential actions", which are probably the thing our users are more aware of. No foreign keys means not referential integrity and referential actions (onDelete, onUpdate), so we offer the `referentialIntegrity = prisma` mode which makes relations possible without and adds emulation of referential actions and integrity.

Might be worthing making that more explicit here, and e.g. call the whole section "Referential actions and integrity, foreign keys".

Reply:
Done.